src/main/java/io/jenkins/plugins/coverage/source/DefaultSourceFileResolver.java

```diff
@@ -105,9 +115,28 @@ private String sanitizeFilename(String inputName) {
         return inputName.replaceAll("[^a-zA-Z0-9-_.]", "_");
     }
 
+    private Map<String, FilePath> createSourceFileMapping(FilePath workspace, TaskListener listener) {
+        try {
+            return Arrays
+                    .stream(workspace.list("**/*"))
```

Comment:
Why list all files in the workspace (which would include compiled binaries, node_modules, etc.) rather than only those referenced by the coverage report(s) being processed? See #150 for something that seems to be caused by this change.

Reply:
If you look at this more closely, you will see that it is executed if the source file we are looking for was not found in the usual way.

And this code resolves source files not only for Cobertura, but also for formats like Jacoco, which does not have such a direct path to the source files.

I'll remove the warnings to don't disturb people

Reply:
Yeah, as a fallback mechanism specifically -- and one that seems to be deliberately using just the filenames without paths -- it doesn't seem worth the noise to have all these duplicates get logged. Otherwise you're logging warnings for a map that might not even get used (and ideally would not, if all went well).

Reply:
I'm looking in the code and don't see this as a fallback, but directly called.  Am I missing something.
I'm investigating why this plugin is taking over 6 hours in CoverageProcessor.convertToResults, while the Cobertura plugin runs in only 21 seconds.  ( I know it's this part of the code per the log out time start of "Publishing Coverage report..." vs the next output of "A total of X reports were found".

Reply:
Yes, that mapping is always constructed -- which may be the cause of your problem -- but is only [_used_](https://github.com/bgreen1005/code-coverage-api-plugin/blob/196f8ce63dcb7cee35d1a90a7dfe8b9673791229/src/main/java/io/jenkins/plugins/coverage/source/DefaultSourceFileResolver.java#L237) if it cannot find a match from the file names/paths in the coverage reports. I believe that's why it was being described as a "fallback".